File: backend/src/main/kotlin/com/casadosreclamos/dto/RequestDto.kt
package com.casadosreclamos.dto

import com.casadosreclamos.model.Request
import com.casadosreclamos.model.request.Measurements
import com.casadosreclamos.model.request.RequestStatus
import com.casadosreclamos.model.request.RequestType
import java.util.*

data class RequestDto(
    var id: Long?,
    var user: String?,
    var client: ClientDto?,
    var created: Date?,
    var lastUpdate: Date?,
    var trackingCode: Long?,
    var status: RequestStatus,
    var type: RequestTypeDto?,
    var cost: Double?,
    var observations: String?,
    var application: Boolean?,
    var brand: BrandDto?
) {
    constructor(request: Request) : this(
        request.id,
        request.requester.name,
        ClientDto(request.client),
        request.created,
        request.lastUpdate,
        request.trackingCode,
        request.status,
        RequestTypeDto.from(request.type!!),
        request.cost,
        request.observations,
        request.application,
        BrandDto(request.brand.id, request.brand.name, null)
    )
}

File: frontend/src/models/RequestDto.js
export default class RequestDto {
  id = null // Long
  user = null // String
  client = null // String
  created = null // Date
  lastUpdate = null // Date
  trackingCode = null // String
  status = null // RequestStatus
  brand = null // BrandDto
  material = null // MaterialDto
  measurements = null // Measurements
  application = null // Boolean
  images = null // RequestTypeDto
  amount = null // Int

  constructor(obj) {
    if (obj) {
      this.id = obj.id
      this.user = obj.user
      this.client = obj.client
      this.created = new Date(obj.created)
      this.lastUpdate = new Date(obj.lastUpdate)
      this.status = obj.status
      this.brand = new BrandDto(obj.brand)
      this.material = new MaterialDto(obj.material)
      this.measurement = new Measurements(obj.measurement)
      this.application = obj.application
      this.images = RequestTypeDto.from(obj.images)
      this.amount = obj.amount
    }
  }
}
